client/executor/common/src/sandbox/wasmer_backend.rs

```diff
@@ -0,0 +1,430 @@
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Wasmer specific impls for sandbox
+
+use crate::{
+	error::Result,
+	sandbox::Memory,
+	util::{checked_range, MemoryTransfer},
+};
+use codec::Encode;
+use sp_core::sandbox::HostError;
+use sp_wasm_interface::{FunctionContext, Pointer, ReturnValue, Value, WordSize};
+use std::{cell::RefCell, collections::HashMap, convert::TryInto, rc::Rc};
+use wasmer::RuntimeError;
+
+use crate::{
+	error::Error,
+	sandbox::{
+		BackendInstance, GuestEnvironment, InstantiationError, SandboxContext, SandboxInstance,
+		SupervisorFuncIndex,
+	},
+};
+
+environmental::environmental!(SandboxContextStore: trait SandboxContext);
+
+/// Wasmer specific context
+pub struct Backend {
+	store: wasmer::Store,
+}
+
+impl Backend {
+	pub fn new() -> Self {
+		let compiler = wasmer_compiler_singlepass::Singlepass::default();
+
+		Backend { store: wasmer::Store::new(&wasmer::JIT::new(compiler).engine()) }
+	}
+}
+
+pub fn invoke_wasmer(
+	instance: &wasmer::Instance,
+
+	// function to call that is exported from the module
+	export_name: &str,
+
+	// arguments passed to the function
+	args: &[Value],
+
+	// arbitraty context data of the call
+	_state: u32,
+
+	sandbox_context: &mut dyn SandboxContext,
+) -> std::result::Result<Option<Value>, wasmi::Error> {
+	let function = instance
+		.exports
+		.get_function(export_name)
+		.map_err(|error| wasmi::Error::Function(error.to_string()))?;
+
+	let args: Vec<wasmer::Val> = args
+		.iter()
+		.map(|v| match *v {
+			Value::I32(val) => wasmer::Val::I32(val),
+			Value::I64(val) => wasmer::Val::I64(val),
+			Value::F32(val) => wasmer::Val::F32(f32::from_bits(val)),
+			Value::F64(val) => wasmer::Val::F64(f64::from_bits(val)),
+		})
+		.collect();
+
+	let wasmer_result = SandboxContextStore::using(sandbox_context, || {
+		function.call(&args).map_err(|error| wasmi::Error::Function(error.to_string()))
+	})?;
+
+	if wasmer_result.len() > 1 {
+		return Err(wasmi::Error::Function("multiple return types are not supported yet".into()))
+	}
+
+	wasmer_result
+		.first()
+		.map(|wasm_value| {
+			let wasmer_value = match *wasm_value {
+				wasmer::Val::I32(val) => Value::I32(val),
+				wasmer::Val::I64(val) => Value::I64(val),
+				wasmer::Val::F32(val) => Value::F32(f32::to_bits(val)),
+				wasmer::Val::F64(val) => Value::F64(f64::to_bits(val)),
+				_ =>
+					return Err(wasmi::Error::Function(format!(
+						"Unsupported return value: {:?}",
+						wasm_value,
+					))),
+			};
+
+			Ok(wasmer_value)
+		})
+		.transpose()
+}
+
+pub fn instantiate_wasmer(
```

Comment:
Same for the other functions and types having wasmer in their name (for the wasmi backend file too). It is already namespaced.

```suggestion
pub fn instantiate(
```